### ISSUE TITLE
Double CRLF is added in start_proxy_connect() when proxy credentials are passed

### ISSUE THREAD

**Issue by dimarusyy** [Microsoft/cpprestsdk#419]
Hello,

The code snippet :	
`if (m_context->m_http_client->client_config().proxy().credentials().is_set())
{
   request_stream << m_context->generate_basic_proxy_auth_header() << CRLF;
}`

But generate_basic_proxy_auth_header() adds CRLF to the end itself :
`utility::string_t generate_basic_proxy_auth_header()
{
....
    header.append(utility::conversions::to_base64(credentials_buffer));
    header.append(CRLF);
    return header;
}`

The issue is reproducible if trying restsdk on linux and doing http request via squid configured for BASIC authentication.

**Comment by leetal** [comment]
Is this causing any connection issues when using credentials? Because another sidenote is that generate_basic_proxy_auth_header() uses the wrong credential-struct as well when connecting to a proxy with basic auth.

The following will fix that on row 808 (develop branch): header.append(generate_base64_userpass(m_http_client->client_config().**proxy()**.credentials()));

**Comment by dimarusyy** [comment]
Yes, that is causing connection issue as squid read's double CRLF treating as HTTP header ending.
I'm fine with the fix you suggested.
